DetoxSync/DetoxSync/Utils/_DTXTimerTrampoline.m

```diff
@@ -112,47 +112,24 @@ - (void)setDisplayLink:(CADisplayLink*)displayLink
 #endif
 }
 
-- (void)fire:(id)timer
+- (void)fire
 {
-	//This is to ensure the timer is still valid after fire.
-	CFRunLoopRef runloop = CFRunLoopGetCurrent();
-	CFRunLoopMode mode = CFRunLoopCopyCurrentMode(runloop);
-	CFRunLoopPerformBlock(runloop, mode, ^{
-		if(CFRunLoopTimerIsValid((__bridge CFRunLoopTimerRef)timer) == NO)
-		{
-			[self untrack];
-			
-			CFRelease(mode);
-			
-			return;
-		}
-		
-		CFRunLoopPerformBlock(runloop, mode, ^{
-			if(CFRunLoopTimerIsValid((__bridge CFRunLoopTimerRef)timer) == NO)
-			{
-				[self untrack];
-				
-				CFRelease(mode);
-				
-				return;
-			}
-			
-			CFRelease(mode);
-		});
-	});
```

Comment:
If I understand right, the purpose of these are to see if the timer is still valid after it has fired.

If we ask if the timer is valid inline here, `CFRunLoopTimerIsValid` always evaluates to true, I assume because we're still within the `fire` function.

Here I'm proposing a different method.

We track the invalidation and removal of timers, which means we know when a repeating timer is stopping. We also know that a timer that doesn't repeat is finished once we have called its callback (or selector).

Therefore, we can simply call `untrack` if we are not a recurring timer after we have called its callback.